packages/react-router/src/components/FocusContextProvider/FocusContextProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React, {useRef, useEffect, ReactNode} from 'react';
+import {FocusContext} from '../../context';
+
+export function FocusContextProvider({children}: {children: ReactNode}) {
+  const focusRef = React.createRef<HTMLElement>();
+  const focus = () => {
+    return focusRef?.current?.focus();
+  };
+
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    if (!firstRender.current) {
+      focus();
+    }
+  });
```

Comment:
Reach has a few optimizations we should probably take:

- Only focus when the URL changes: https://github.com/reach/router/blob/master/src/index.js#L319-L323
- Something something avoiding problems with `autofocus`: https://github.com/reach/router/blob/master/src/index.js#L341-L345

Reply:
> Only focus when the URL changes: 

Done ✅ 

> Something something avoiding problems with autofocus 

It looks like we should be okay with the current implementation here. Autofocus is only supposed to apply on initial page load/render (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefautofocus, https://github.com/facebook/react/issues/5534#issuecomment-158972163). Since we're not forcing focus on the initial load, `autofocus` will still work with this implementation.